### PR TITLE
Add suction gripper as an option for the Oberon 7 arm

### PR DESCRIPTION
**Pull request by mjcarroll**
Integrates @arjo129 's suction gripper into the oberon arm launcher.

To spawn with the suction gripper (for ease of verification, I have been spawning it on dry land).

```
ros2 launch mbzirc_ign spawn.launch.py name:=usv world:=coast model:=usv type:=usv x:=-1495 y:=0 z:=4.3 R:=0 P:=0 Y:=0 arm:=mbzirc_oberon7_arm gripper:=mbzirc_suction_gripper
```

This exposes two ROS topics:

* `/<MODEL NAME>/arm/gripper/contact` - List of contacts with the end effector
* `/<MODEL NAME>/arm/gripper/suction_on` - Used to control if the gripper is running

For example:
```
# Disable the gripper
ros2 topic pub --once /usv/arm/gripper/suction_on std_msgs/msg/Bool 'data: False'  

# Enable the gripper
ros2 topic pub --once /usv/arm/gripper/suction_on std_msgs/msg/Bool 'data: False'  
```

Showing picking up a very light unit cube and dropping it:
ezgif-1-517e4a4e8b

### CODE REVIEW

File: mbzirc_ign/src/mbzirc_ign/bridges.py [hunk 1]
```diff
@@ -169,3 +169,22 @@ def arm_camera_info(world_name, model_name, link_name):
         ign_type='ignition.msgs.CameraInfo',
         ros_type='sensor_msgs/msg/CameraInfo',
         direction=BridgeDirection.IGN_TO_ROS)
+
+def gripper_suction_contacts(model_name):
```
Comment (iche033) [inline mbzirc_ign/src/mbzirc_ign/bridges.py:173]: hmm I think we probably don't want to bridge this topic. The contact data contains a scoped name of the collision that the gripper comes into contact with, which lets the user uniquely identify the object

We could either filter out the collision fields or just provide one force torque sensor to give some feedback that the suction gripper has pressed onto something.

File: mbzirc_ign/test/helper/TestFixture.hh [hunk 2]
```diff
@@ -83,9 +83,16 @@ class MBZIRCTestFixture : public ::testing::Test
   /// \param[in] World file name.
   public: void LoadWorld(std::string _world)
   {
+    std::string foo;
+    ignition::common::env("IGN_GAZEBO_RESOURCE_PATH", foo);
+
+    std::cout << foo << std::endl;
```
Comment (iche033) [inline mbzirc_ign/test/helper/TestFixture.hh:89]: remove?

File: mbzirc_ign/test/helper/TestFixture.hh [hunk 3]
```diff
@@ -83,9 +83,16 @@ class MBZIRCTestFixture : public ::testing::Test
...
+
     auto worldPath = ignition::common::joinPaths(
       std::string(PROJECT_SOURCE_PATH), "worlds", _world);
 
+    std::cout << worldPath << std::endl;
```
Comment (iche033) [inline mbzirc_ign/test/helper/TestFixture.hh:94]: remove?

**Review by iche033 — COMMENTED** [review]
works for me. I have a comment on bridging the contact topic

File: mbzirc_ign/test/helper/TestFixture.hh (hunk 2 above)
Comment (mjcarroll) [reply mbzirc_ign/test/helper/TestFixture.hh:89]: Yep, leftover from cleaning up the environment hooks.

File: mbzirc_ign/src/mbzirc_ign/bridges.py (hunk 1 above)
Comment (mjcarroll) [reply mbzirc_ign/src/mbzirc_ign/bridges.py:174]: I have entirely removed the topic for now.

**Comment by iche033** [comment]
I made a small tweak to the suction visual material to help tell the outer and inner cylinders apart
mbzirc_suction_gripper

Also added a `break` on the `OnContact` as I think it shouldn't need to iterate over all objects in contact as we we'll just create a joint with one other model. 2227f5ef55478cf83c95c605725b522f7cd339b0

**Comment by iche033** [comment]
I noticed that the RTF dropped from 8x% to 6x% with the contact sensor in the world. I think it's due to extra processing done in the physics system here:
https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo6/src/systems/physics/Physics.cc#L3175

I think we should be able to optimize this in the future.

**Review by iche033 — APPROVED** [review]
works for me

**Pull request merged**